io/directory.go

```diff
@@ -342,7 +370,16 @@ func (d *HAMTDirectory) Find(ctx context.Context, name string) (ipld.Node, error
 
 // RemoveChild implements the `Directory` interface.
 func (d *HAMTDirectory) RemoveChild(ctx context.Context, name string) error {
```

Comment:
This logic needs to be extended to `AddChild` which can actually overwrite an entry and cause a size reduction.

Blocking.

Reply:
https://github.com/ipfs/go-unixfs/pull/103